google/resource_bigquery_dataset.go

```diff
@@ -180,6 +226,45 @@ func resourceDataset(d *schema.ResourceData, meta interface{}) (*bigquery.Datase
 		dataset.Labels = labels
 	}
 
+	if v, ok := d.GetOk("access"); ok {
+		access := []*bigquery.DatasetAccess{}
+		for _, m := range v.([]interface{}) {
+			da := bigquery.DatasetAccess{}
+			msi := m.(map[string]interface{})
+			da.Role = msi["role"].(string)
+			if val, ok := msi["domain"]; ok {
```

Comment:
I would need to confirm 100% since I don't remember which cases this applies to and which it doesn't, but I _think_ the map will actually have all values in it, even if they weren't set, which would mean you could do:
```
da := bigquery.DatasetAccess{
  Role: msi["role"].(string),
  Domain: msi["domain}.(string),
  [etc]
}
```

Reply:
Really? You're already inside a block that checks whether access is set once you've gotten to this point.